src/index.js

```diff
@@ -197,7 +208,7 @@ class CID {
    * @returns {string}
    */
   toBaseEncodedString (base) {
-    base = base || 'base58btc'
+    base = base || this.base
```

Comment:
My JS-foo is not good enough. Is

```
base = base || this.base
^^^^---- this
```
the `let base` from the constructor?

Reply:
```js
x = x || 'some default'
```
is a old js idiom for setting a default value where one is not given.

In this case `base` referers only to the parameter from method signature

``` js
toBaseEncodedString (base) {
```

as it would shadow any definition of `base` from an outer scope.

I believe we're using ES6 features in a bunch of places now, so this could be simplifed to 

```js
toBaseEncodedString (base = this.base) {
```

https://css-tricks.com/using-default-parameters-es6/

Reply:
Thanks for the detailed explanation, I knew all that but wasn't able to connect the dots.